From 5193f5cebb214286c54818ac81a0ea3d1823ecaf Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 25 Sep 2020 02:01:38 -0700
Subject: [PATCH 1672/1851] ASoC: xlnx: sdiaudio: Fix warnings for DEADCODE

commit fa0277b029e122638480748e485e7c00fb9f8c83 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the coverity warnings for the DEADCODE type.

The warnings in xlnx_sdi_rx_pcm_startup() are

Event dead_error_condition: The "switch" governing value "val &
CHAN_ID_0" cannot be "2U".
Event dead_error_begin:	Execution cannot reach this statement:
"case 2U:".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_sdi_audio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/xilinx/xlnx_sdi_audio.c b/sound/soc/xilinx/xlnx_sdi_audio.c
index badea10b6e16..961502c46122 100644
--- a/sound/soc/xilinx/xlnx_sdi_audio.c
+++ b/sound/soc/xilinx/xlnx_sdi_audio.c
@@ -189,7 +189,7 @@ static int xlnx_sdi_rx_pcm_startup(struct snd_pcm_substream *substream,
 
 	val = ioread32(base + XSDIAUD_EXT_SRATE_STS_REG_OFFSET);
 	/* As both channels contain same sample rate, read either of them */
-	switch (val & CHAN_ID_0) {
+	switch (val & CHAN_ID_1) {
 	case 0:
 		sample_rate = 48000;
 		break;
-- 
2.31.1

